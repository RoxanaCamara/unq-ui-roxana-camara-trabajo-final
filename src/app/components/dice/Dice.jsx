import React from "react";
import PropTypes from 'prop-types';
import "./style.css";

const Dice = ({ num }) => {
  return (
    <>
      <div className="nono">
        {num == 1 && (
          <div className="first-face">
            <span className="pip"></span>
          </div>
        )}
        {num == 2 && (
          <div className="second-face">
            <span className="pip"></span>
            <span className="pip"></span>
          </div>
        )}
        {num == 3 && (
          <div className="third-face">
            <span className="pip"></span>
            <span className="pip"></span>
            <span className="pip"></span>
          </div>
        )}
        {num == 4 && (
          <div className="fourth-face">
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
          </div>
        )}
        {num == 5 && (
          <div className="fifth-face">
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
            <div className="column">
              <span className="pip"></span>
            </div>
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
          </div>
        )}
        {num == 6 && (
          <div className="sixth-face">
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
            <div className="column">
              <span className="pip"></span>
              <span className="pip"></span>
              <span className="pip"></span>
            </div>
          </div>
        )}
      </div>
    </>
  );
};


Dice.propTypes = {
  num: PropTypes.number.isRequired,
};

export default Dice;
