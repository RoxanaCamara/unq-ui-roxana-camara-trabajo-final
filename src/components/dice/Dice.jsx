import React, { useContext, useState } from 'react'
import { Grid } from '@material-ui/core';
import '../dice/style.css'
import Checkbox from '@material-ui/core/Checkbox';
import { SessionContext } from '../helper/Session';


export const Dice = ({ num, index }) => {

    const { actions, state } = useContext(SessionContext)
    const { dadosGuardados } = state
    const { setDadosGuardados } = actions
    const [checked, setChecked] = useState(false);


    const handleChange = (event) => {
      let tem = [...dadosGuardados]
      if(event.target.checked){
          tem[index] = num
      }
      else{
        tem[index] = 0
      }
      setChecked(event.target.checked);
      setDadosGuardados(tem)
    };
    return (
        <>
        <div className="nono">
        <Grid item xs={4}>
        {
            num == 1 &&
            <div className="first-face">
                <span className="pip"></span>
            </div>
        }
        {
            num == 2 &&
            <div className="second-face">
                <span className="pip"></span>
                <span className="pip"></span>
            </div>
        }
        {
            num == 3 &&
            <div className="third-face">
                <span className="pip"></span>
                <span className="pip"></span>
                <span className="pip"></span>
            </div>
        }
        {
            num == 4 &&
            <div className="fourth-face">
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
            </div>
        }
        {
            num == 5 &&
            <div className="fifth-face">
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
                <div className="column">
                    <span className="pip"></span>
                </div>
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
            </div>
        }
        {
            num == 6 &&
            <div className="sixth-face">
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
                <div className="column">
                    <span className="pip"></span>
                    <span className="pip"></span>
                    <span className="pip"></span>
                </div>
            </div>
        }
        <Checkbox checked={checked} onChange={handleChange} inputProps={{ 'aria-label': 'primary checkbox' }} />  
        </Grid> 
        </div>
        </>
    )
}
