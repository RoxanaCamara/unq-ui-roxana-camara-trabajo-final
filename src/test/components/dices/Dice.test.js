import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Dice from '../../../app/components/dice/Dice';

describe('Test del component <Dice/>', () => {

    const wrapper = shallow( <Dice num={ 4 } />);

    test('Debemos testear cuando reciba un numero del 1 al 6', () => {
        
        //expect( wrapper ).toMatchSnapshot();  
        console.log(wrapper.html())
        const p = wrapper.find('#fourth-face');
        console.log(p)
        //expect( p.text().trim() ).toBe( title );
        
       
        //expect( className.includes('fourth-face') ).toBe( true ); 
    })

    test('Debemos testear cuando reciba un numero mayor a 6', () => {
        //const wrapper = shallow( <Dice num={ 8 } /> );
        /*const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('fourth-face') ).toBe( false );*/ 

    })
})