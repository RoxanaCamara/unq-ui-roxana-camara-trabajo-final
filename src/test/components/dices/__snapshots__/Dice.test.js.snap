// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del component <Dice/> Debemos testear cuando reciba un numero del 1 al 6 1`] = `
<Fragment>
  <div
    className="fourth-face"
  >
    <div
      className="column"
    >
      <span
        className="pip"
      />
      <span
        className="pip"
      />
    </div>
    <div
      className="column"
    >
      <span
        className="pip"
      />
      <span
        className="pip"
      />
    </div>
  </div>
</Fragment>
`;
